### PR TITLE
Generate configuration documentation from ci

### PR DESCRIPTION
**Pull request by Yaraslaut**
Refs https://github.com/contour-terminal/contour/issues/1622

Adds reflection-cpp dependency and handles generation of global configuration part of the documentation

### CODE REVIEW

**Review by christianparpart — APPROVED** [review]
Only minor comments. Many thanks. :)

**Comment by christianparpart** [comment]
p.s.: clang-tidy complains a lot about unused parameters in Config.cpp

File: src/contour/ContourApp.cpp [hunk 1]
```diff
@@ -246,6 +248,121 @@ int ContourApp::documentationKeyMapping()
...
+
+!!! note "Please note that on Unix systems, the environment variable `XDG_CONFIG_HOME` (by default set to `~/.config`) is taken into account."
+
+By default, on Unix systems, Contour is executed with the following arguments `contour config ~/.config/contour/contour.yml profile main`. If the configuration file includes a `default_profile` variable, it will be used as the default profile. Otherwise, the first profile listed in the file will be the default one.
+## How to
+
+### Load specific configuration file
+`contour config /path/to/file/with/configuration.yml`
+### Set profile for current session
+you can utilize the `profile` parameter with the `contour` command <br/>
```
[the remark this answers is not in the document]
Comment (Yaraslaut) [reply src/contour/ContourApp.cpp:277]: This is for web-page rendering: https://contour-terminal.org/configuration/#how-to

File: src/contour/CMakeLists.txt [hunk 2]
```diff
@@ -213,6 +213,8 @@ else()
 endif()
 
 target_link_libraries(contour
+    PUBLIC
```
[the remark this answers is not in the document]
Comment (Yaraslaut) [reply src/contour/CMakeLists.txt]: Deleted this part, link to crispy instead and then it is propagated well

**Pull request merged**